### PR TITLE
feat: block non github domain and drawer enhancement

### PR DESCRIPTION
**Pull request by KKzLEO**

### CODE REVIEW

**Review by shinenic — CHANGES_REQUESTED** [review]
Maybe we choose `optional permission` for enterprise GitHub instead of matching all route.

**Comment by KKzLEO** [comment]
> Maybe we choose `optional permission` for enterprise GitHub instead of matching all route.

Sure, now we're using chrome extension's context menu to control enabling/disabling a-tree in specific domain, but we still need to set `matches` to `all_urls`

File: src/background.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
-import { GLOBAL_MESSAGE_TYPE } from 'constants'
-
-const activeTabUrlMemo = {}
```
Comment (shinenic) [inline src/background.js:3]: I think we still need this one.

File: src/background.js [hunk 2]
```diff
@@ -21,3 +19,26 @@ chrome.tabs.onRemoved.addListener((tabId) => {
     delete activeTabUrlMemo[tabId]
   }
 })
+
+chrome.runtime.onInstalled.addListener(() => {
```
Comment (shinenic) [inline src/background.js:23]: Cool!!
btw, maybe we can add the contextMenu option only on `non-GitHub domain` pages?

File: public/manifest.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "manifest_version": 3,
   "homepage_url": "https://github.com/shinenic/a-tree",
   "minimum_chrome_version": "60",
-  "permissions": ["webNavigation"],
+  "permissions": ["webNavigation", "contextMenus"],
```
Comment (shinenic) [inline public/manifest.json:8]: Nice work, don't forget to update `README.md`

File: src/background.js (hunk 2 above)
Comment (shinenic) [reply src/background.js:23]: not sure if it is possible (or necessary

File: src/background.js (hunk 1 above)
Comment (shinenic) [reply src/background.js:3]: or the SPA will not work lol

File: src/index.js [hunk 4]
```diff
@@ -5,8 +5,55 @@ import { CONTAINER_ID } from 'constants'
...
+const checkDomainMatched = (domains) => {
+  const host = window.location.host
+
+  return domains.includes(host) || host === 'github.com'
+}
+
+const applyStyleFromLocalStorage = (drawerWidth) => {
+  const style = document.createElement('style')
+  style.innerHTML = `
+    body {
```
Comment (shinenic) [inline src/index.js:51]: `html` for Mac?

**Pull request merged**